src/accelerate/checkpointing.py

```diff
@@ -196,12 +196,12 @@ def load_accelerator_state(
         ending = f"_{i}" if i > 0 else ""
         input_model_file = input_dir.joinpath(f"{SAFE_MODEL_NAME}{ending}.safetensors")
         if input_model_file.exists():
-            state_dict = load_file(input_model_file, device=str(map_location))
+            load_model(model, input_model_file, device=str(map_location), **load_model_func_kwargs)
         else:
             # Load with torch
             input_model_file = input_dir.joinpath(f"{MODEL_NAME}{ending}.bin")
             state_dict = torch.load(input_model_file, map_location=map_location)
-        models[i].load_state_dict(state_dict, **load_model_func_kwargs)
+            model.load_state_dict(state_dict, **load_model_func_kwargs)
     logger.info("All model weights loaded successfully")
```

Comment:
Any particular reason for this change? I'd expect only the prior to be modified.

Reply:
In the if statement, he's loading the safetensors model directly whereas before, we were only getting the state dict.

Reply:
_load_model_ does both: it loads the file and uses it to populate the _state_dict_. Previously, each branch of the if-condition only loaded the file and after the if-condition, the model would load the state dict. Since _load_model_ does both, I indented the statement on line 204 to become part of the else-clause. This becomes clearer when you have a look at the complete surroundings of the changes instead of only the affected lines.